pkg/environment/env.go

```diff
@@ -179,17 +184,64 @@ func (env *Environment) downloadComponent(component string, version pkgver.Versi
 
 // SelectInstalledVersion selects the installed versions and the latest release version
 // will be chosen if there is an empty version
-func (env *Environment) SelectInstalledVersion(component string, version pkgver.Version) (pkgver.Version, error) {
-	return env.profile.SelectInstalledVersion(component, version)
-}
-
-// DownloadComponentIfMissing downloads the specific version of a component if it is missing
-func (env *Environment) DownloadComponentIfMissing(component string, ver pkgver.Version) (pkgver.Version, error) {
-	versions, err := env.profile.InstalledVersions(component)
+func (env *Environment) SelectInstalledVersion(component string, ver pkgver.Version) (pkgver.Version, error) {
+	installed, err := env.Profile().InstalledVersions(component)
 	if err != nil {
 		return "", err
 	}
 
+	errInstallFirst := errors.Annotatef(ErrInstallFirst, "use `tiup install %s` to install component `%s` first", component, component)
```

Comment:
why not use `ErrInstallFirst` here?

Reply:
Because this error will be displayed to the user